compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/PrefetchSharedMemoryCopy.cpp

```diff
@@ -176,9 +176,54 @@ class LoopPrefetcher {
     });
   }
 
+  void getValueDependenciesForIf(scf::IfOp ifOp,
+                                 DenseSet<Operation *> &readDependencies,
+                                 DenseSet<Operation *> &writeDependencies) {
+    // scf.if with results should be supported directly through the usual
+    // handling so bail-out here in that case
+    if (ifOp->getNumResults() != 0)
+      return;
+    bool hasGlobalRead = false;
+    bool hasSharedWrite = false;
+    // Else region not yet supported.
+    if (!ifOp.getElseRegion().empty()) {
+      return;
+    }
+    ifOp->walk([&](Operation *op) {
+      if (auto readOp = dyn_cast<vector::TransferReadOp>(op)) {
```

Comment:
There are operation other that transfer_read that might be of interest? How do we know there won't be a transfer_gather in here?

Reply:
This pass doesnt consider ops other than read/write, see [here](https://github.com/iree-org/iree/pull/20904/files#diff-691ac54fdc87fce9c6d084cc646aa4e0fd617bc617767191a9e568ebe0ff2d2bL187) so not much point in adding that handling in the specific case of if they are present in an if. What should happen is that there should be a counterpart op to the gather outside the if and that wont find a stage and the pass will bail-out.